legacy/billing/src/main/java/com/appcoins/wallet/billing/common/BillingErrorMapper.kt

```diff
@@ -32,51 +32,51 @@ open class BillingErrorMapper @Inject constructor(private val gson: Gson) {
 
   private fun getErrorType(
     httpCode: Int?, messageCode: String?,
-    data: Any?
+    data: String?
   ): ErrorInfo.ErrorType {
     return when {
       messageCode == NOT_ALLOWED_CODE -> ErrorInfo.ErrorType.SUB_ALREADY_OWNED
       messageCode == FORBIDDEN_CODE -> ErrorInfo.ErrorType.BLOCKED
       httpCode == CONFLICT_HTTP_CODE -> ErrorInfo.ErrorType.CONFLICT
-      messageCode == ADYEN_V2_ERROR && (data is Number) -> {
-        when (data.toInt()) {
-          101 -> {
-            ErrorInfo.ErrorType.INVALID_CARD
+      messageCode == ADYEN_V2_ERROR && (data != null) -> {
+        when (data.toString()) {
+          "14_029" -> {
+            ErrorInfo.ErrorType.CVC_REQUIRED
           }
 
-          103 -> {
+          "103" -> {
             ErrorInfo.ErrorType.CVC_LENGTH
           }
 
-          105 -> {
+          "105" -> {
             ErrorInfo.ErrorType.CARD_SECURITY_VALIDATION
           }
 
-          138 -> {
+          "138" -> {
             ErrorInfo.ErrorType.CURRENCY_NOT_SUPPORTED
           }
 
-          200 -> {
+          "200" -> {
             ErrorInfo.ErrorType.INVALID_COUNTRY_CODE
           }
 
-          172, 174, 422, 800 -> {
+          "172", "174", "422", "800", "803" -> {
```

Comment:
I think it would be more clear with a enum, magic numbers could be very hard to understand in the future, and, if possible, group some of them.

Reply:
Might be better to leave this error codes like this. They come like this from Adyen and each represent the error from the documentation.